doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,195 @@
+
+# Pre-requisite
+
+As mentioned above in 'Background' and 'Plan' sections, need to follow specified bring-up sequence.
+Work flows are designed considering SONiC NOS operating in sync mode.
+
+In case SONiC NOS operates in async mode, then expected behavior is - the return status of the set ADMIN_STATE attribute update in ASIC-DB (syncd/GBsyncd) will be treated to set the host_tx_ready in Orchagent.
+
+# Breakout Handling
+  - The new 'host_tx_ready' field of Port table in state-DB is created for every interface <regular/breakout interface>.
+  - Xcvrd processes the 'host_tx_ready' change event and is responsible to disable Tx/laser for all optical lanes or respective optical lane that belongs to the interface in case of breakout.
+  - Currently the logical mapping between the interface and optical lane is not present in xcvrd. Creating this logical mapping in xcvrd will address breakout interface handling.
+
+# Proposed Work-Flows
+
+Please refer to the  flow/sequence diagrams which covers the following required use-cases
+  - Enabling this feature 
+  - Transceiver initialization
+  - admin enable configurations 
+  - admin disable configurations
+  - No transceiver present
+
+# Feature enablement
+  This feature (optics Interface Link bring-up sequence) would be enabled on per platform basis.
+  There could be cases where vendor(s)/platform(s) may take time to shift from existing codebase to the model (work-flows) described in this document.
+  In order to avoid any breakage and ensure gradual migration of different platforms/vendors to this model, there would be new field (flag) in xcvrd to enable/disable this feature. 
+  When xcvrd spawns on LC/board, it would invoke platform plugin to check with the platform (hwsku) whether this feature is yet supported on underlying platform (board/LC) or not
+  
+  Workflow :
+  ![Enabling 'Interface link bring-up sequence' feature(3)](https://user-images.githubusercontent.com/69485234/152266723-050377ce-d4de-4c67-a405-5acc66474d46.png)
+
+
+# Transceiver Initialization 
+  (at platform bootstrap layer)
+  
+![LC boot-up sequence - optics INIT (platform bootstrap)](https://user-images.githubusercontent.com/69485234/152261613-e20dcda9-2adc-42aa-a1f1-4b8a47dd32af.png)
+
+# Applying 'interface admin startup' configuration
+
+![LC boot-up sequence - 'admin enable' Config gets applied](https://user-images.githubusercontent.com/69485234/147166867-56f3e82d-1b1c-4b7a-a867-5470ee6050e7.png)
+
+
+# Applying 'interface admin shutdown' configuration
+
+![LC boot-up sequence - 'admin disable' Config gets applied](https://user-images.githubusercontent.com/69485234/147166884-92c9af48-2d64-4e67-8933-f80531d821b4.png)
+
+# No transceiver present
+if transceiver is not present:
+ - All the workflows mentioned above will reamin same ( or get exercised) till host_tx_ready field update
+ - xcvrd will not perform any action on receiving host_tx_ready field update 
```

Comment:
will xcvrd perform the actions when receiving optical module plugin event later on?

Reply:
@keboliu 
>> The xcvrd 'host_tx_ready' field event handler will ignore the change of event when no optical module present. 
Xcvrd will always refer for the host_tx_ready field in state-db to perform the DP-Init/activate during the module initialization when an optical module is inserted.